src/main/python/systemds/context/systemds_context.py

```diff
@@ -164,4 +164,14 @@ def rand(self, rows: int, cols: int, min: Union[float, int] = None,
         :param lambd: lamda value for "poison" distribution
         :return:
         """
+        available_pdfs = ["uniform", "normal", "poisson"]
+        if rows < 0:
+            raise ValueError("In rand statement, can only assign rows a long (integer) value >= 0 "
+                            "-- attempted to assign value: {r}".format(r=rows))
+        if cols < 0:
+            raise ValueError("In rand statement, can only assign cols a long (integer) value >= 0 "
```

Comment:
`if cols <= 0:`
`In rand statement, can only assign cols a long (integer) value > 0 `

not >=0

This makes me think that you did not verify the edge case of 0. Other than that, the message looks good :smile:

Reply:
I did not include the zero case because `np.random.randint(0,1,(0,0))` is a valid call. And `sds.rand(0,0)` returns the same output (an empty numpy array). Should I change the condition anyways?

Reply:
hmm, then duplicate the behavior from NP, and see if it creates an empty output. How does DASK do it? also empty array?

Reply:
Just to finish this thread, you have done the correct thing! :100:, I did not consider that it is useful to make code that have a 0x0 matrix to start with, and then add columns and rows to that, iteratively.